From 8d6bc0e8ab9ae8b9245b0dd1686f36521caf668b Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 3 Mar 2016 22:41:24 -0500
Subject: [PATCH] wr-policy: allow radiusd_t getcap to itself

* Allow radiusd_t get linux capabilities of itself (process)

Fix avc denial:

  avc: denied { getcap } for pid=1196 comm="radiusd" \
  scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
  tcontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023 \
  tclass=process permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/services/radius.te | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/policy/modules/services/radius.te b/policy/modules/services/radius.te
index e6ff2d0..1488b4f 100644
--- a/policy/modules/services/radius.te
+++ b/policy/modules/services/radius.te
@@ -34,7 +34,7 @@ files_pid_file(radiusd_var_run_t)
 
 allow radiusd_t self:capability { chown dac_override fsetid kill setgid setuid sys_resource sys_tty_config };
 dontaudit radiusd_t self:capability sys_tty_config;
-allow radiusd_t self:process { getsched setrlimit setsched sigkill signal };
+allow radiusd_t self:process { getcap getsched setrlimit setsched sigkill signal };
 allow radiusd_t self:fifo_file rw_fifo_file_perms;
 allow radiusd_t self:unix_stream_socket { accept listen };
 allow radiusd_t self:tcp_socket { accept listen };
@@ -99,6 +99,7 @@ fs_search_auto_mountpoints(radiusd_t)
 
 files_read_usr_files(radiusd_t)
 files_read_etc_runtime_files(radiusd_t)
+files_map_etc_files(radiusd_t)
 files_dontaudit_list_tmp(radiusd_t)
 
 auth_use_nsswitch(radiusd_t)
-- 
2.7.4

